src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -286,12 +286,6 @@ public void OnStopActivity(Activity activity, object payload)
                 // the one created by the instrumentation.
                 // And retrieve it here, and set it to Current.
             }
-
-            var textMapPropagator = Propagators.DefaultTextMapPropagator;
-            if (textMapPropagator is not TraceContextPropagator)
-            {
-                Baggage.Current = default;
```

Comment:
should this be done at SDK side instead? Could you run stress tests to confirm there are no leaks if we do not remove it at all.

Reply:
I did run stress test. Did not see any leaks.

Reply:
> should this be done at SDK side instead?
Are you referring to OnEnd in SDK? I have not looked in to that but just thinking it will impact all activities as we need to first check the activity ending is ASP.NET Core one.

Reply:
Baggage should be independent of any activity, so why we need to check which activity is being stopped?

Reply:
Only for ASP.NET Core activity. For other one's we do not do anything right now as well.